web/src/message_edit.js

```diff
@@ -399,13 +400,10 @@ function create_copy_to_clipboard_handler($row, source, message_id) {
     });
 
     clipboard.on("success", () => {
-        end_message_row_edit($row);
-        $row.find(".alert-msg").text($t({defaultMessage: "Copied!"}));
-        $row.find(".alert-msg").css("display", "block");
-        $row.find(".alert-msg").delay(1000).fadeOut(300);
-        if ($(".tooltip").is(":visible")) {
-            $(".tooltip").hide();
-        }
+        show_copied_confirmation($row.find(".copy_message")[0]);
+        setTimeout(() => {
+            end_message_row_edit($row);
+        }, 1000);
```

Comment:
This is OK but it's possible that a cleaner implementation would be to have pass `end_message_row_edit($row);` as a `on_hide_callback` callback to `show_copied_confirmation`, and have that be responsible for calling the callback in its `setTimeout` handler -- the benefit being that we wouldn't be relying on an unchecked assertion that 1000 is the same timeout here and there.

Reply:
Great suggestion! That's up now in the updated PR.